rover/source/docs/sailboat-hardware.rst

```diff
@@ -19,10 +17,61 @@ to the wind speed the more important this is. See :ref:`wind-vane`.
-type wind speed sensor two more analogue inputs are required, although just a single additional input can be used with a
-small loss in accuracy. Alternatively a pressure based airspeed sensor can be used (link to optional hardware airspeed).
-Ideally a sail vehicle autopilot would have three analogue inputs. Pixhawk1 and the CUBE both fit this criteria.
+The servo must pull in and let out the sails in a symmetrical way. I.e. no change of servo position should be required to set the sails at the same angle on a new tack. This is typically 
+achieved through the use of a sail winch servo with a main sheet system. The sail winch is servo output function, ``SERVOx_FUNCTION``, 89 - Main Sail.
+
+When using a sail winch the sail limits angle must be configured. 
+This can be tested by arming in manual mode; the throttle will directly control the sail position.
+Throttle stick down (towards you) should result in the sail being sheeted in towards the centre line.
+If the sail is sheeted out the servo should be reversed. The servo min and max parameters can then be
+used to set up the range of travel. The min and max values should be set such that the boom is brought
+in towards the centre line of the boat but not pulled down tightly. The boom should be able to be let
+out until it reaches the shrouds; if no shrouds are fitted the boom shouldn’t go too far past 90 degrees
+to the boat centre line. 
+
+The sail angle :ref:`SAIL_ANGLE_MIN<SAIL_ANGLE_MIN>` and :ref:`SAIL_ANGLE_MAX<SAIL_ANGLE_MAX>` parameters should be set
+to the angle to boom is to the centre line at each extreme of its travel. This allows the angle of the
+boom to be calculated at any point between. 
+
+The ideal sail angle to the wind :ref:`(SAIL_ANGLE_IDEAL)<SAIL_ANGLE_IDEAL>` should then be set. This defines the angle
+between the boom and the wind direction as reported by the wind vane. An angle of zero here would result
+in the boom staying parallel to the wind vane. The boom will keep this angle to the wind until it reaches
+either its minimum or maximum limit. If the sails are too loose this number should be increased. This can
+be tested in ACRO mode. 
+
+Wing sail elevator
+------------------
+A wing sail may be actuated by a secondary control surface akin to a plane's elevator, the wing is free to rotate on a mast and actuated only by the control surface. This should be setup such that the servo mid point (``SERVOx_TRIM``) results in zero angle of attack on the main wing.
+See `Karl's build <https://discuss.ardupilot.org/t/sailboat-support/32060/321>`__ . Use servo output function, ``SERVOx_FUNCTION``, 128 - Wing Sail Elevator.
+
+The servo min (``SERVOx_MIN``) and max (``SERVOx_MAX``) end points should be set such that the main wing is held in a maximum lift angle of attack on both tacks.
+If the wing sail is acting on the wrong tack the servo should be reversed with ``SERVOx_REVERSE``. The `SAIL_ANGLE_*` parameters are not use with wing sail elevators.
+
```

Comment:
Trying to recall whether these instructions remains the same with the patch we looked at to change the wing sail PID control: https://github.com/IamPete1/ardupilot/tree/wing_sail_pid.

I'm fairly sure the parameters didn't need to change, but the proposed controller would allow the wing elevator to reverse in order to counter heel when cornering. Probably best to update documentation for that behaviour at the time that patch is merged.

Reply:
@srmainwaring Do you think that branch is enough of a improvement to consider bringing into main AP?